File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        //   int cnt = 0;
        //   set<int>st;
        //   vector<int>ans;

        //  for(int i=0;i<nums.size();i++){
        //     if(st.find(nums[i]) == st.end()){
        //         st.insert(nums[i]);
        //     }
           

        //  } 
        //  int i=0;
        //  for(auto it : st){
        //    nums[i] = it;
        //    i++;
        //  }
        //  return st.size();
        int i=0;
     for(int j=1;j<nums.size();j++){
          if(nums[i] != nums[j]){
             i++;
            nums[i] = nums[j];
           
          }
     }
         return i+1;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int st =0; int end = 0; int cnt =0;
        while(st<nums.length){
          while(end<nums.length && nums[st]==nums[end])end++;

          if(end == nums.length)break;
          st++;
          nums[st] = nums[end];
          cnt++;
        }
       return cnt+1;
    }
}